includes/classes/Command.php

```diff
@@ -747,11 +746,21 @@ function ( $prefix ) use ( $args ) {
 		$query_args = [];
 
 		$query_args['offset'] = 0;
-
 		if ( ! empty( $args['offset'] ) ) {
 			$query_args['offset'] = absint( $args['offset'] );
 		}
 
+		// More performant pagination. Keeping behind this feature flag for now during testing.
+		if ( ! empty( $args['advanced-pagination'] ) ) {
+			$query_args['ep_indexing_advanced_pagination'] = true;
+
+			if ( ! empty( $args['start-post-id'] ) ) {
+				$query_args['ep_indexing_start_post_id'] = absint( $args['start-post-id'] );
+				// Perpetually remember for the "total_objects" count to keep the progress bar accurate.
+				$query_args['ep_indexing_initial_start_post_id'] = absint( $args['start-post-id'] );
```

Comment:
Not sure I follow. So every round there is a progress message like: `Processed 500/300,000`. We get that 300,000 from the first query that runs with `SQL_CALC_FOUND_ROWS` present.

Say there are 250,000 posts that need to be indexed, and the CLI is passed with `--start-post-id=100000`. The first loop will say `Processed 500/150,000`, the next loop - `Processed 1000/150,000`. 

It needs to remember that initial `start-post-id` that is passed in so that the first `SQL_CALC_FOUND_ROWS` query that tells us how many posts are being indexed during this run is always the same (and thus cached).

Reply:
Ohh I see, I read this comment as "use this value as total_objects count" and got confused :)

Reply:
Switching the logic to be based around a moving `ep_indexing_last_processed_object_id` value instead. So `start_post_id` will always stay the same based on initial input - thus no more need for `ep_indexing_initial_start_post_id`.